### PR TITLE
chore(pf-icons): create icon definition and generate pf icons

### PR DESCRIPTION
**Pull request by karelhala**
* Introduce new way of how to define icons
* Add all pf-icons currently supported

resolves: https://github.com/patternfly/patternfly-next/issues/730

### CODE REVIEW

**Comment by karelhala** [comment]
I am not sure about the lock file, it's really huge so I don't really know if this is correct way of pushing it, but I had to install new package called `node-plop` and lock it in certain version.

**Comment by karelhala** [comment]
I don't know how to ignore paths for gatsby so I moved the icons definition and everything under `src` not sure if that's good way though.

File: package.json [hunk 1]
```diff
@@ -10,8 +10,9 @@
     "backstop:approve": "node test/approve-conflicts.js",
     "backstop:init": "node test/create-references.js",
     "bootstrap:build": "node npm-scripts/build-bootstrap-workspace.js",
-    "build-patternfly": "rm -rf dist && rm -rf tmp && gulp build",
+    "build-patternfly": "rm -rf dist && rm -rf tmp && npm run build:icons && gulp build",
     "build": "gatsby build",
+    "build:icons": "node ./src/icons/generateIcons.js",
```
Comment (dgutride) [inline package.json]: File paths are case sensitive - please change to src/icons in the actual name to be consistent with the rest of our naming.

File: src/Icons/PfIcons/add-circle-o-icon.svg [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (dmiller9911) [inline src/Icons/PfIcons/add-circle-o-icon.svg]: Might better to only check-in the Json file and just have these generated at build. That way the Json file doesn't get out of sync.

Comment (karelhala) [reply src/Icons/PfIcons/add-circle-o-icon.svg]: Mhhh, yeah that makes sence, I'll work on it.

File: package.json (hunk 1 above)
Comment (karelhala) [reply package.json]: Probly a problem when I moved stuff around.

Comment (priley86) [reply package.json]: +1 - i think if we can resolve this and rename to `/src/Icons/generateIcons` path the PR is acceptable now.

File: src/Icons/PfIcons/add-circle-o-icon.svg (hunk 2 above)
Comment (priley86) [reply src/Icons/PfIcons/add-circle-o-icon.svg]: Filed https://github.com/patternfly/patternfly-next/issues/771. Let's revisit this later, would like to unblock the PR if possible!

**Pull request merged**